### ISSUE TITLE
corrupted string in certain pattern

### ISSUE THREAD

**Issue by leesei** [maxogden/ndjson#17]
I'm parsing UTF8 files in Chinese (multiple bytes per character) with NDJSON, some strings in the parsed JSON are corrupted. Yet the same string appearing on other lines are fine.
When I copied the corrupted lines out to a smaller file (which fits in one buffer on `data`, the problem disappeared.
It seems to be a bug of buffer concatenation in the old `split2` (`^0.2.1`) and `through2` (`^0.6.1`) package `ndjson` is using.

I switch to the latest version (2.15) and the error does not occur.
```js
fs.createReadStream(argv.input)
  //.pipe(ndjson.parse())
  .pipe(split2(JSON.parse))
  .pipe(through2.obj(function (data, enc, callback) {
      console.log(data);
    });
    callback();
  }));
```

**Comment by finnp** [comment]
Thank you :) I published this with updated deps at v1.4.4. Let me know if it worked out for you.

**Comment by leesei** [comment]
Confirm fixed in 1.4.4 and 1.5.
Thx.
